emission/tests/modellingTests/TestForestModelIntegration.py

```diff
@@ -0,0 +1,132 @@
+        self.has_label_percent = 0.9
+        ## generate mock trips
+        train = etmm.generate_mock_trips(
+            user_id=self.testUUID,
+            trips=self.total_trips,
+            origin=self.origin,
+            destination=self.destination,
+            trip_part='od',
+            label_data=label_data,
+            within_threshold=self.clustered_trips,  
+            threshold=0.004, # ~400m
+            has_label_p=self.has_label_percent
+        )
+        ts.bulk_insert(train)
+        # confirm data write did not fail
+        check_data = esda.get_entries(key="analysis/confirmed_trip", user_id=self.testUUID, time_query=None)
+        if len(check_data) != self.total_trips:
+            logging.debug(f'test invariant failed after generating test data')
+            self.fail()
+        else:
+            logging.debug(f'found {self.total_trips} trips in database')
+        ## Build an already existing model or a new model
+        eamur.update_trip_model(
+            user_id=self.testUUID,
+            model_type=eamumt.ModelType.RANDOM_FOREST_CLASSIFIER,
+            model_storage=eamums.ModelStorage.DOCUMENT_DATABASE,
+            min_trips=4,
+            model_config=forest_model_config
+        )
+        ## run inference pipeline
+        self.run_pipeline(self.test_algorithms)
+        time_range = estt.TimeQuery("metadata.write_ts", None, time.time())
+        self.inferred_trips = esda.get_entries(esda.INFERRED_TRIP_KEY, self.testUUID, time_query=time_range)
+
+    def tearDown(self):
+        self.reset_all()
+
+    def run_pipeline(self, algorithms):
+        default_primary_algorithms = eacilp.primary_algorithms
+        eacilp.primary_algorithms = algorithms
+        epi.run_intake_pipeline_for_user(self.testUUID,skip_if_no_new_data = False)
+        eacilp.primary_algorithms = default_primary_algorithms
+
+    def reset_all(self):
+        etc.dropAllCollections(edb._get_current_db())
+
+    # Tests that forest algorithm being tested runs successfully
+    def testForestAlgorithm(self):
+        for trip in self.inferred_trips:
+            entries = esdt.get_sections_for_trip("inference/labels", self.testUUID, trip.get_id())
+            self.assertEqual(len(entries), len(self.test_algorithms))
+            for entry in entries:
+                self.assertGreater(len(entry["data"]["prediction"]), 0)
+                for singleprediction in entry["data"]["prediction"]:
+                    self.assertIsInstance(singleprediction, dict, " should be an instance of the dictionary class")
+                    self.assertIsInstance(singleprediction['labels'], dict, " should be an instance of the dictionary class")
+                    self.assertIn('mode_confirm',singleprediction['labels'].keys())
+                    self.assertIn('replaced_mode',singleprediction['labels'].keys())
+                    self.assertIn('purpose_confirm',singleprediction['labels'].keys())        
+
```

Comment:
> This just checks that the properties exist, not that they are correct. 

The test here ensures that the model has the required properties to ensure it  can make predictions without errors. Correctness checking of the predictions would require hard coding these predictions in tests which might change as the model evolves ( even for better). In such a case, even if the model improved, the test would fail ( since the predictions changed). 

>When will the "more and more tests" from [#938 (comment)](https://github.com/e-mission/e-mission-server/pull/938#discussion_r1381828612) be implemented?

Right now we have test to check if the model loads and saves correctly, if it builds correctly, runs correctly and if it integrates in the pipeline as expected.

 I am unable to think of anymore tests  from my limited knowledge. Can we think brainstorm and think what more can be added?